src/main/java/use_cases/play_card_use_case/PlayCardInteractor.java

```diff
@@ -0,0 +1,198 @@
+            player.playCard(cardId);
+            player.setStructure((StructureCard) card);
+            PlayCardOutputModel outputModel = getOutputModel();
+            return playCardPresenter.updateGameScreen(outputModel);
+        }
+        // The only other case is the card being Action, so we can safely cast it.
+        ActionCard action = (ActionCard) card;
+        List<Integer> targetIds = new ArrayList<>();
+        switch (action.getTargetType()) {
+            case SINGLE_ENEMY:
+                targetIds.addAll(getCreatureIds(opponent.getCreatures()));
+                break;
+            case SINGLE_FRIENDLY:
+                targetIds.addAll(getCreatureIds(player.getCreatures()));
+                break;
+            case ANY:
+                targetIds.addAll(getCreatureIds(player.getCreatures()));
+                targetIds.addAll(getCreatureIds(opponent.getCreatures()));
+                break;
+        }
+        if (targetIds.size() != 0) {
+            TargetModel targetModel = new TargetModel(cardId, targetIds);
+            return playCardPresenter.showTargetSelectionScreen(targetModel);
+        }
+
+        // If we didn't return in the above, we must be arriving at an action
+        // that can be immediately spent. So we *spend* it from the player's hand
+        // first, so that any cards drawn would not mistakenly get discarded
+        player.playCard(cardId);
+        switch (action.getTargetType()) {
+            case FRIENDLY:
+                useAction(action, player.getCreatures());
+                break;
+            case ENEMY:
+                useAction(action, opponent.getCreatures());
+                break;
+            case ALL:
+                useAction(action, player.getCreatures());
+                useAction(action, opponent.getCreatures());
+                break;
+            case SELF:
+                useAction(action, player);
+                break;
+            case OPPONENT:
+                useAction(action, opponent);
+                break;
+
+        }
+        PlayCardOutputModel outputModel = getOutputModel();
+        return playCardPresenter.updateGameScreen(outputModel);
+    }
+
+    /**
+     * Spend the card that the player chose to play, and trigger its effects
+     * on the target given by {@code targetId}.
+     * The target will always be a creature, so we can safely assume
+     * the effects are CreatureStatsEffects.
+     *
+     * @param cardId   the id of the card being played.
+     * @param targetId the id of the creature to be affected.
+     * @return a PlayCardOutputModel, containing all possible changed stats in
+     * the gameState.
+     */
+    @Override
+    public PlayCardOutputModel playSingleTargetCard(int cardId, int targetId) {
+        Player player = gameState.getCurrentPlayer();
+        Player opponent = gameState.getOpposingPlayer();
+
+        ActionCard card = (ActionCard) player.playCard(cardId);
+
```

Comment:
I really got screwed by this <playerCard> method (LOL), because when I checked this method above in this file, I thought this method should not return anything that can be downcasted to an ActionCard object.
I think those two different <playCard> methods can be distinguished by the object they are facing to. The method in the Player.java file is used when a player chooses a card and decides to let it roll in the game pool. Certainly, <playCard> is an appropriate name. However, the other one which is in the PlayCardInteractor.java looks like some procedures inside the code. 

Overall, I think <engageCard> sounds more friendly to the reader.

Reply:
yeah the name and signature of the interactor method is the same as the player's method, we'll change the names for clarity later.